File: backend/slotify/spring/src/main/java/com/slotify/backend/spring/reserve/dtos/ReserveSummary.java
package com.slotify.backend.spring.reserve.dtos;

import com.fasterxml.jackson.annotation.JsonSubTypes;
import com.fasterxml.jackson.annotation.JsonTypeInfo;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.experimental.SuperBuilder;

import java.time.LocalDateTime;
import java.util.UUID;
@Getter
@SuperBuilder
@NoArgsConstructor
@JsonTypeInfo(
        use = JsonTypeInfo.Id.NAME,
        include = JsonTypeInfo.As.PROPERTY,
        property = "summaryType",
        visible = true
)
@JsonSubTypes({
        @JsonSubTypes.Type(value = UserReserveSummary.class, name = "USER"),
        @JsonSubTypes.Type(value = CompanyReserveSummary.class, name = "COMPANY")
})
public abstract class ReserveSummary {
    private UUID reserveId;
    private UUID serviceId;

    private LocalDateTime startDateTime;
    private LocalDateTime endDateTime;
    private Integer minutesDuration;

    private String serviceName;
    private Integer servicePriceCent;

    private Boolean isCanceled;
}

File: frontend/app/src/main/java/com/raj/slotify/dtos/reserves/ReserveSummary.kt
package com.raj.slotify.dtos.reserves

import android.os.Parcelable
import kotlinx.parcelize.Parcelize
import java.time.LocalDateTime
import java.util.UUID

@Parcelize
sealed class ReserveSummary : Parcelable {
    abstract val reserveId: UUID
    abstract val serviceId: UUID
    abstract val startDateTime: LocalDateTime
    abstract val endDateTime: LocalDateTime
    abstract val minutesDuration: Int
    abstract val serviceName: String
    abstract val servicePriceCent: Int
    abstract val isCanceled: Boolean
}